infra/resources/prod/main.tf

```diff
@@ -132,3 +132,23 @@ module "iam" {
 
   cdn_storage_account_id = module.cdn.storage_account_cdn.id
 }
+
+module "apim" {
+  source = "../_modules/apim"
+
+  project_legacy = local.project_legacy
+  apim = {
+    name                = local.apim.name
```

Comment:
Can we use a `data` to ensure we have the right references?

Reply:
The `name` and `resource group` are essential values for constructing the data for `APIM`. Given that these values are already required for this purpose, define a data to retrieve the same two outputs might be redundant and unnecessary.